File: runtime/python/units/rt/fceil.py
# requires: rt/quiet_nan
@staticmethod
def fceil(x):
    if x != x:
        return Rt.quiet_nan(x)
    if not math.isfinite(x):
        return x
    if x == 0.0:
        return x
    r = math.ceil(x)
    return -0.0 if (r == 0 and x < 0.0) else float(r)

File: runtime/perl/units/rt/fceil.pl
# requires: rt/quiet_nan
# C99 ceil handles infinities and signed zeros; NaNs are quieted by hand
# because the spec requires it and the C round-trip may not.
use POSIX ();
sub fceil {
    my $x = $_[0];
    return Rt::quiet_nan($x) if $x != $x;
    return POSIX::ceil($x);
}
